src/meta/src/stream/stream_graph/fragment.rs

```diff
@@ -649,21 +680,38 @@ impl CompleteStreamFragmentGraph {
                                             .map(|i| i as u32)
                                     })
                                     .collect::<Option<Vec<_>>>()
-                                    .context("column not found in the upstream materialized view")?
+                                    .context(
+                                        "column not found in the upstream materialized view",
+                                    )?;
+                                (dist_key_indices, output_indices)
+                            };
+                            let dispatch_strategy = if uses_arrangement_backfill {
+                                if !dist_key_indices.is_empty() {
```

Comment:
What about determining strategy with `fragment.distribution_type`?

Reply:
Hmm the `fragment` here is a `BuildingFragment`, which does not have such a field. Is there some other way to get this field? I see that `Fragment` has it.